src/api.rs

```diff
@@ -49,3 +55,35 @@ pub async fn package_rate(
     };
     (Status::Ok, Either::Left(Json(ret)))
 }
+
+#[delete("/reset")]
+pub async fn package_reset(mod_db: &State<ModuleDB>) -> (Status, &'static str) {
+    let mut write_lock = mod_db.write().await;
+    write_lock.clear();
+    for module in write_lock.values_mut() {
```

Comment:
You cleared the hashmap... then you read the values in it? I suggest `drain()`, which clears the map and returns an iterator.